@6.agl.guppy

antiapuentes opened the pull request:
Contains work to remove the dependency on Qt and some changes specific to AGL.

Commit:
Make PmLog optional

Change-Id: Ic61d3c67ac290ef763944605cb70d33bc6735804
Reviewed-by: Jani Hautakangas <jani.hautakangas@lge.com>
Tested-by: Jani Hautakangas <jani.hautakangas@lge.com>

Commit:
[agl][build][luna-service] Make luna service optional

Change-Id: Ia3996fbfb500abbe0570790b5aa465c553dc8248
Reviewed-by: Jani Hautakangas <jani.hautakangas@lge.com>
Tested-by: Jani Hautakangas <jani.hautakangas@lge.com>

Commit:
[agl][build][webos] Guard WebOS specific parts

Change-Id: I10f32bcfe441638ec0ac0013bf4f6e35f57a1b14
Reviewed-by: Jani Hautakangas <jani.hautakangas@lge.com>
Tested-by: Jani Hautakangas <jani.hautakangas@lge.com>

Commit:
[agl][build] Hardcode MACHINE_NAME for now

Change-Id: I88fc660860dc9e88ed967064c6c0615ef4c5a216
Reviewed-by: Jani Hautakangas <jani.hautakangas@lge.com>
Tested-by: Jani Hautakangas <jani.hautakangas@lge.com>

Commit:
Add empty stub for agl service integration

Change-Id: I040cb32a5774f84926dbab869d9c869662672dda
Reviewed-by: Jani Hautakangas <jani.hautakangas@lge.com>
Tested-by: Jani Hautakangas <jani.hautakangas@lge.com>

Commit:
[agl][build] Add --launch-app parameter

Change-Id: Ieb3ebd27d1e2a925461ccc7857f577d21c8589ce
Reviewed-by: Jani Hautakangas <jani.hautakangas@lge.com>
Tested-by: Jani Hautakangas <jani.hautakangas@lge.com>

Commit:
[agl][build] Parse launch app appinfo.json

Change-Id: Ie32e8246abe57eaf15eeb68601c7e90ca2cc300b
Reviewed-by: Jani Hautakangas <jani.hautakangas@lge.com>
Tested-by: Jani Hautakangas <jani.hautakangas@lge.com>

Commit:
[agl][build] Trigger load visually committed if not OS_WEBOS

Temporary patch to workaround splashscreen that's expected
by wam on webos

Change-Id: I89ee92dc94c12767fd3f6857293659bd4e374729
Reviewed-by: Jani Hautakangas <jani.hautakangas@lge.com>
Tested-by: Jani Hautakangas <jani.hautakangas@lge.com>

Commit:
Read config.xml instead fo appinfo.json

Change-Id: Iab1f2241f225c6be4b78f23c869689cd71f3f964
Reviewed-by: Jose Dapena <jose.dapena@lge.com>
Tested-by: Jose Dapena <jose.dapena@lge.com>

andre-rosa committed:
[agl][build] Add systemd startup script and environment

Change-Id: I81237c47ba9374ca797309404dc279878ab84103
Reviewed-by: Jose Dapena <jose.dapena@lge.com>
Tested-by: Jose Dapena <jose.dapena@lge.com>

Commit:
[agl] Separate wam host and client instances

Host:
/usr/bin/WebAppMgr --no-sandbox --user-data-dir="/home/root/wamdata" --webos-wam

Client:
/usr/bin/WebAppMgr --launch-app /var/local/lib/afm/applications/webapps-memory-match/1.1

Change-Id: I595090a078d318d70e4bbee259c3cb6e9a9f0ba7
Reviewed-by: Jani Hautakangas <jani.hautakangas@lge.com>
Tested-by: Jani Hautakangas <jani.hautakangas@lge.com>

Commit:
[agl] Set the window surface ID later

:Release Notes:
Set the window surface ID later

:Detailed Notes:
Launching apps would crash, because the surface ID was set before
the WebAppWindowBase was created. So set it after it is created.

:Testing Performed:

:QA Notes:

:Issues Addressed:

Change-Id: I08d8533138a38fff33865b5df3c8ddffd157508d
Reviewed-by: Jose Dapena <jose.dapena@lge.com>
Tested-by: Jose Dapena <jose.dapena@lge.com>

Commit:
[agl] Set the IVI pid at the right time.

Change-Id: I2359696dbf1b904d19b04c1f0b42f22a8ebeabf8
Reviewed-by: Jose Dapena <jose.dapena@lge.com>
Tested-by: Jose Dapena <jose.dapena@lge.com>

andre-rosa committed:
[agl] Update wam host instance startup command

Change-Id: Iaf91b7810a180998edd3c101c30c7b806bdee79c
Reviewed-by: Andre Rosa <andre.rosa@lge.com>
Tested-by: Andre Rosa <andre.rosa@lge.com>
Reviewed-by: Jose Dapena <jose.dapena@lge.com>
Tested-by: Jose Dapena <jose.dapena@lge.com>

Commit:
[agl] Separate service lock and socket implementation

Change-Id: I2cd4f8fa2b8c91f9de9b247fbd2ffcb887cfa708
Reviewed-by: Jani Hautakangas <jani.hautakangas@lge.com>
Tested-by: Jani Hautakangas <jani.hautakangas@lge.com>

Commit:
[agl] Add support for wm/hs 2017

Commit:
[agl] Add support for homescreen app switching

Also refactors code to reduce the mess in Main.cpp

Commit:
[agl] Add support for shared browser process model

andre-rosa committed:
Use weaker Wants instead of BindsTo dependency on weston

See https://www.freedesktop.org/software/systemd/man/systemd.unit.html

Change-Id: I5e8f473dde1c7ffe6348965946a7c12699aa93e5

sebd71 committed:
Used new WindowManager APIs

WindowManager APIs have changed:
 - activateSurface => activateWindow
 - deactivateSurface => deactivateWindow

See https://jira.automotivelinux.org/browse/SPEC-1565

Change-Id: If80be16a6af3e33df870e0ebc85b4c6f8f41a2c3
Signed-off-by: Sebastien Douheret <sebastien.douheret@iot.bzh>

jdapena committed:
[WIP] Launch tinyproxy

Before launching a webapp through shared browser process model, current
implementation launch tinyproxy using -p <port_number> commandline option
and forwaed same port to WebViewProfile to configure proxyservice

Co-authored-by: Abhijeet Kandalkar <abhijeet@igalia.com>

abhijeetk committed:
Launch webapp using tinyproxy.

Implement proxy configuration handling for shared browser process
WebAppLauncher

jkim-julie committed:
Update user directory

It updates user directory path for flounder from
"/home/root/" to "/home/0/"

jkim-julie committed:
Add RestartSec

hferreiro committed:
Store proxy rules in WebView profiles

Previously, the default profile was being used for all web apps. As a
result of this, the proxy rules where being shared by all web apps, and
the same tinyproxy process would be used by all of them. SMACK would
then block tinyproxy requests for all web apps but the first one, which
would be the one that configured the SMACK label in the tinyproxy
process.

Now, a mapping between app ids and WebView profiles is stored and
applied to each WebView. The proxy rules are stored on those profiles.

[SPEC-1895] error launching web app: "tinyproxy was unable to connect to the remote web server"
https://jira.automotivelinux.org/browse/SPEC-1895

nickdiego committed:
[agl] Fix indentation issues

Fix minor coding style indentation issues.

Signed-off-by: Nick Diego Yamane <nickdiego@igalia.com>

nickdiego committed:
[agl][qtless] Do not use QJson in PalmSystem and WebApp

- Modify PalmSystemBase, PalmSystemWebOS, PalmSystemBlink,
  WebAppBase and WebAppWayland classes removing QJson* references
  in favor of a Jsoncpp-based implementation.

[SPEC-1910] Qt-less WAM: replace QJson classes usage in WAM
[SPEC-1907] Qt-less WAM: do not use Qt JSON

Signed-off-by: Nick Diego Yamane <nickdiego@igalia.com>

nickdiego committed:
[agl][qtless] Add initial JsonHelper implementation

- Initial implementation of Jsoncpp helper functions.
  These functions abstract and streamline the json parsing
  from both strings and files, in addition they make it possible
  to handle potential errors in a unified and standardized way.
- This patch is part of Qt-less WAM effort, which aims to remove
  the WAM dependency on Qt framework.

[SPEC-1910] Qt-less WAM: replace QJson classes usage in WAM
[SPEC-1907] Qt-less WAM: do not use Qt JSON

Signed-off-by: Nick Diego Yamane <nickdiego@igalia.com>

nickdiego committed:
[agl][qtless] Do not use QJson in ApplicationDescription

- Modify ApplicationDescription class removing QJson* references
  in favor of a Jsoncpp-based implementation.

[SPEC-1910] Qt-less WAM: replace QJson classes usage in WAM
[SPEC-1907] Qt-less WAM: do not use Qt JSON

Signed-off-by: Nick Diego Yamane <nickdiego@igalia.com>

nickdiego committed:
[agl][qtless] Do not use QJson in ContainerAppManager

- Modify ContainerAppManager class removing QJson* references
in favor of a Jsoncpp-based implementation.

[SPEC-1910] Qt-less WAM: replace QJson classes usage in WAM
[SPEC-1907] Qt-less WAM: do not use Qt JSON

Signed-off-by: Nick Diego Yamane <nickdiego@igalia.com>

nickdiego committed:
[agl][qtless] Do not use QJson in WebAppFactoryManager

- Modify WebAppFactoryManager removing QJson* references
in favor of a Jsoncpp-based implementation.

[SPEC-1910] Qt-less WAM: replace QJson classes usage in WAM
[SPEC-1907] Qt-less WAM: do not use Qt JSON

Signed-off-by: Nick Diego Yamane <nickdiego@igalia.com>

nickdiego committed:
[agl][qtless] Do not use QJson in WebPageBase

- Modify WebPageBase class removing QJson* references
in favor of a Jsoncpp-based implementation.

[SPEC-1910] Qt-less WAM: replace QJson classes usage in WAM
[SPEC-1907] Qt-less WAM: do not use Qt JSON

Signed-off-by: Nick Diego Yamane <nickdiego@igalia.com>

nickdiego committed:
[agl][qtless] Do not use QJson in WebAppManager and WebProcessManager

- Modify WebAppManager, WebProcessManager, BlinkWebProcessManager
and NetworkStatus classes removing QJson* references in favor of a
Jsoncpp-based implementation.

[SPEC-1910] Qt-less WAM: replace QJson classes usage in WAM
[SPEC-1907] Qt-less WAM: do not use Qt JSON

Signed-off-by: Nick Diego Yamane <nickdiego@igalia.com>

nickdiego committed:
[agl][qtless] Do not use QJson in WebAppManagerService

- Modify WebAppManagerService and WebAppManagerServiceAGL classes
removing QJson* references in favor of a Jsoncpp-based implementation.

[SPEC-1910] Qt-less WAM: replace QJson classes usage in WAM
[SPEC-1907] Qt-less WAM: do not use Qt JSON

Signed-off-by: Nick Diego Yamane <nickdiego@igalia.com>

nickdiego committed:
[agl][qtless] Drop WebAppFactory pluggable system

As part of Qt-less effort, this patch drops WAM's Qt-based
Pluggable WebAppFactory system (at least temporarily) for AGL
build. To achieve it, (if agl_service build flag is set) we now
compile WebAppFactoryManagerAGL and WebAppFactoryAGL classes and
link them against libWebAppMngrCore and libWebAppMngr in order
to statically handle WebApps with "default" app type.

[SPEC-1913] Qt-less WAM: drop Qt plugin usage
[SPEC-1871] [meta] Qt-less WAM

Signed-off-by: Nick Diego Yamane <nickdiego@igalia.com>

hferreiro committed:
Change deprecated include <assert.h> to <cassert>

andre-rosa committed:
Start wam as systemd user service correcting its dependencies

wam systemd scripts were waiting for weston to launch. But this is
not enough, as for proper integration with AGL window manager and
homescreen it is using those services. So wam should wait for those
services.

[SPEC-1944] WAM: service fails to start upon very first boot after flashing
https://jira.automotivelinux.org/browse/SPEC-1944

hferreiro committed:
Delete the webapp profile mapping when closed

When a webapp is closed, cleanup the WebViewProfile that was created to
store that particular webapp proxy rules.

[SPEC-1933] delete the WebViewProfile/proxy rules mapping when an app is closed
https://jira.automotivelinux.org/browse/SPEC-1933

nickdiego committed:
[agl][jsoncpp] Better handle empty JSON strings

It's pretty common empty JSON strings to be passed in
to WAM's json handling code. In former Qt-based implementation
that was handled silently. To get the same behavior when using
Jsoncpp we need extra code otherwise Json::Reader implementation
raises runtime exceptions and some cases even print error messages
(1.8.3, for example). This patch add this extra code making it
painless to parse empty JSON string in WAM.

[SPEC-1941] WAM: jsoncpp shows warnings opening applications in log

Signed-off-by: Nick Diego Yamane <nickdiego@igalia.com>

nickdiego committed:
[agl][jsoncpp] Fix sub-objects serialization

When serializing nested objects from a JSON object
QJson behaves differently from Jsoncpp when that field
is not present in the root object. Qt returns an empty
object (string "{}") while Jsoncpp return a null value
(string "null"). This patch modifies ApplicationDescription
implementation to result in empty JSON object strings even
when that field is not present.

[SPEC-1941] WAM: jsoncpp shows warnings opening applications in log

Signed-off-by: Nick Diego Yamane <nickdiego@igalia.com>

nickdiego committed:
[agl][qtless] Cleanup Q_SLOTS usage in WebPage classes

Remove unused/obsolete Q_SLOT functions from WebPageBase

[SPEC-1912] Qt-less WAM: drop QObject + Q_SLOT + Q_SIGNALS

Signed-off-by: Nick Diego Yamane <nickdiego@igalia.com>

nickdiego committed:
[agl][qtless] Drop signal/slots in favor of WebPageObserver

- This patch replaces Qt-specific signals/slots usage in WAM
by an Observer pattern based approach. Such mechanism is used
to route WebView events between WebPage and WebApp objects.
- Assuming all events coming from underlying WebView implementation
are called in the WAM main loop's thread, this observer-based approach
should be equivalent to Qt-based signal/slot previous implementation.

[SPEC-1912] Qt-less WAM: drop QObject + Q_SLOT + Q_SIGNALS

Signed-off-by: Nick Diego Yamane <nickdiego@igalia.com>

nickdiego committed:
[agl][qtless] Replace Qt data container types by their STL counterpars

- This patch replaces Qt data structures, such as QMap, QPair, QList
by std::map, std::pair, and std::list/vector/set where applicable.

[SPEC-1911] Qt-less WAM: replace qt basic data structures

Signed-off-by: Nick Diego Yamane <nickdiego@igalia.com>

nickdiego committed:
[agl][qtless] Migrate ApplicationDescription away from QString

This patch get rid of QString/QStringList usage from
ApplicationDescription implementation and some related APIs.

[SPEC-1911] Qt-less WAM: replace qt basic data structures

Signed-off-by: Nick Diego Yamane <nickdiego@igalia.com>

nickdiego committed:
[agl][qtless] Add StringUtils initial implementation

StringUtils is intended to fill the gaps between
C++ standard library and Qt API's regarding string
manipulation. E.g: Qt provides a split() function
out-of-the-box while std library does not.

[SPEC-1911] Qt-less WAM: replace qt basic data structures

Signed-off-by: Nick Diego Yamane <nickdiego@igalia.com>

nickdiego committed:
[agl][qtless] Get rid of QString usage in WebProcessManager

- This patch removes QString usage from WebProcessManager implementation.
To do that it also modifies classes that have some relationship with it,
such as WebAppManager and WebAppManagerService.
- Also, add some FIXME comments to mark points for next steps on removing
QString references from WAM.

[SPEC-1911] Qt-less WAM: replace qt basic data structures

Signed-off-by: Nick Diego Yamane <nickdiego@igalia.com>

jkim-julie committed:
Replace |m_pid_v| with |m_pid_map|

It replaces |m_pid_v| with |m_pid_map|. As WAM uses one browser
process, WAM gets the same value from |surf_props.creatorPid| at
notify_ivi_control_cb and it can not recognize a unique ID with
|surf_props.creatorPid|. So, there was a hack to replace it with
|id| instead of |surf_props.creatorPid|. In order to consider
sharing the code with runxdg from [SPEC-1920], this change replaces
the data structure for |m_pid_v| with |m_pid_map|.

[SPEC-1920] Refactor libwindowmanager and libhomescreen out of WAM
https://jira.automotivelinux.org/browse/SPEC-1920

nickdiego committed:
[agl][qtless] Migrate WebAppManager away from QString

- This patch removes QString usage from WebAppManager, focusing
on removing its usage almost entirely from its public API, to do
that it also modifies classes that have some relationship with it,
such as WebApp(Base and Wayland), WebAppManagerService, WebAppFactory,
etc.

[SPEC-1911] Qt-less WAM: replace qt basic data structures

Signed-off-by: Nick Diego Yamane <nickdiego@igalia.com>

nickdiego committed:
[agl][qtless] Do not use QString WebAppManager in JS strings building

This patch focus on removing QString usage in Javascript code generation
in WebApp initialization, also touching WebPageBase and WebPageBlink
implementation, modifying its 'evaluateJavascript()' method to receive
an std::string instead of QString.

[SPEC-1911] Qt-less WAM: replace qt basic data structures

Signed-off-by: Nick Diego Yamane <nickdiego@igalia.com>

nickdiego committed:
[agl][qtless] QString-less WebAppManager: SystemLanguage and DeviceInfo

This patch removes QString usage from SystemLanguage and DeviceInfo
API's in WebAppManager, WebAppManagerService, DeviceInfo, WebPageBase
and WebPageBlink.

[SPEC-1911] Qt-less WAM: replace qt basic data structures

Signed-off-by: Nick Diego Yamane <nickdiego@igalia.com>

nickdiego committed:
[agl][qtless] QString-less WebApp/WebPage: appId

This patch removes QString usage from appId APIs in WebApp and WebPage
classes. Devided to keep this as a separate patch because appId() heavy
usage all over WAM code.

[SPEC-1911] Qt-less WAM: replace qt basic data structures

Signed-off-by: Nick Diego Yamane <nickdiego@igalia.com>

nickdiego committed:
[agl][qtless] QString-less WebApp: Remove leftovers

Remove QString leftovers from WebAppBase and WebAppWayland,
besides that modifies interfaces from related classes.

[SPEC-1911] Qt-less WAM: replace qt basic data structures

Signed-off-by: Nick Diego Yamane <nickdiego@igalia.com>

nickdiego committed:
[agl][qtless] Migrate WebAppWayland away from QVariant

This patch refactors WebAppWayland::setWindowProperty() method, removing
its dependency on QVariant. As most of its used was using QString and in
the end the value gets convered to a std::string the solution adopted
was to modify its parameter to a std::string.

[SPEC-1911] Qt-less WAM: replace qt basic data structures

Signed-off-by: Nick Diego Yamane <nickdiego@igalia.com>

nickdiego committed:
[agl][qtless] Do not use QString in WebPage

[SPEC-1911] Qt-less WAM: replace qt basic data structures

Signed-off-by: Nick Diego Yamane <nickdiego@igalia.com>

nickdiego committed:
[agl][qtless] Do not use QString in NetworkStatus

[SPEC-1911] Qt-less WAM: replace qt basic data structures

Signed-off-by: Nick Diego Yamane <nickdiego@igalia.com>

nickdiego committed:
[agl][qtless] Do not use QString in WebAppManagerConfig

[SPEC-1911] Qt-less WAM: replace qt basic data structures

Signed-off-by: Nick Diego Yamane <nickdiego@igalia.com>

nickdiego committed:
[agl][qtless] Migrate PalmSystem away from QString

[SPEC-1911] Qt-less WAM: replace qt basic data structures

Signed-off-by: Nick Diego Yamane <nickdiego@igalia.com>

nickdiego committed:
[agl][qtless] QString-less WAM: Remove leftovers

[SPEC-1911] Qt-less WAM: replace qt basic data structures

Signed-off-by: Nick Diego Yamane <nickdiego@igalia.com>

nickdiego committed:
[agl][qtless] Add Url parsing code

- The main goal of this patch is to remove WAM dependency on QUrl
implementation for Url Parsing and convenience API. To do that, we're
adding our own Url API, which in this initial version leverages GLib URI
Functions [1] for basic URI parsing and escaping.

- Also, as the code handling URLs in WAM is mostly tied to
QFile/QDir/QFileInfo manipulation, this patch also refactor that to use
Boost Filesystem module instead. It's worth noting it depends on a
modified version os meta-agl-lge (which include boost as dependency) to
build properly. This integration is still very primitive and will be
improved soon once we have WAM build system ported from qmake to CMake.

[SPEC-1911] Qt-less WAM: replace qt basic data structures

Signed-off-by: Nick Diego Yamane <nickdiego@igalia.com>

nickdiego committed:
[agl][qtless] Remove QVariant leftovers

- This patch removes QVariant usage from PalmSystem* and
WebAppManagerConfig classes.
- Completely remove PalmSystemBase::getResource() and
WebAppManagerConfig::get/setConfiguration() since they are not being
used anywhere else.
- Modify PalmSystemWebOS's pmLog* functions that were expecting
QVariants as parameter to receive std::strings instead, since they were
being used only with QStrings.
- Anyway, if we really need those methods/variables in the future we can
use boost::any to replace QVariant basic functionality.

[SPEC-1911] Qt-less WAM: replace qt basic data structures

Signed-off-by: Nick Diego Yamane <nickdiego@igalia.com>

nickdiego committed:
[agl][qtless] Replace QLocale by Boost.Locale

- This patch removes QLocale usage from WAM. It was being
used in WebPageBlink, to generate search path for localized
error pages.
- The previous implementation assumed QLocale::bcp47Name()
function returns an IETF Language Tag (BCP 47) string for
language string got from DeviceInfo API. Even though after
some tests with Qt API, it seems that bcp47Name() is not being
including 'script' field even when it's passed in the input string,
this path implements a boost-based genPathForLang() function that
returns a similar formatted path assuming the input string follows
the format decribed in QLocale documentation [1]

[1] http://doc.qt.io/qt-5/qlocale.html#QLocale-1

[SPEC-1911] Qt-less WAM: replace qt basic data structures

Signed-off-by: Nick Diego Yamane <nickdiego@igalia.com>

nickdiego committed:
[agl][qtless] Trim timestamp string in NetworkStatus

[SPEC-1911] Qt-less WAM: replace qt basic data structures

Signed-off-by: Nick Diego Yamane <nickdiego@igalia.com>

nickdiego committed:
[agl][qtless] Use const char arrays instead of macros for WindowTypes

[SPEC-1911] Qt-less WAM: replace qt basic data structures

Signed-off-by: Nick Diego Yamane <nickdiego@igalia.com>

hferreiro committed:
[DROP][agl][chromium68] Pass the correct app_runtime::WebViewProfile

Temporary hack while we fix the chromium68 APIs

nickdiego committed:
[WIP][agl] Improve debug/logging messages

Signed-off-by: Nick Diego Yamane <nickdiego@igalia.com>

nickdiego committed:
[agl][chromium68] Disable gpu process

nickdiego committed:
[agl][chromium68] Fix crash when setting proxy server

- Use empty strings for user and password instead of 'nullptr'
  when set proxy information. This fixes a crash when launching
  web app in WAM/chromium68.

Signed-off-by: Nick Diego Yamane <nickdiego@igalia.com>

msisov committed:
[agl] Register only one surface.

All the surfaces created by wam have the same pid, which
results in a wrong launcher using surfaces and activating it.

Most of the time it results in an error message and whenever
the broadcast message is received by a right launcher, the surface
is requested and activated.

A right approach will be to compare associated surface id, which
wam created for that app launcher.

msisov committed:
[agl] React on Inactive and Active events

We need to pass to WebAppMgr host process the events, which launcher
receives. At this point, we need Inactive and Active events.

The only quick way to do it is by sending a messages through a socket
and calling the methods in WebAppManagerServiceAGL in a right
message loop through timer.

This makes the playing content to be stopped and avoid wasting
resources.

antiapuentes committed:
[agl] Use ShowWindow event instead of TapShortcut

[SPEC-2154] Invisible layer created for WAM applications

antiapuentes committed:
[agl] use appid instead of appname in tap_shortcut

[SPEC-2154] Invisible layer created for WAM applications

nickdiego committed:
[agl][qtless] Remove Qt dependency from qmake files

For now, Qt::Core is kept as dependency on non-AGL (WebOSOSE) build os
WAM. That dependency will be removed in a followup patch.

Signed-off-by: Nick Diego Yamane <nickdiego@igalia.com>

nickdiego committed:
[agl][qtless] Get rid of Qt usage in webos/DeviceInfoImpl

Signed-off-by: Nick Diego Yamane <nickdiego@igalia.com>

nickdiego committed:
[agl][qtless] Port to CMake

TODO: Verify/fix webos build

Signed-off-by: Nick Diego Yamane <nickdiego@igalia.com>

nickdiego committed:
[agl][qtless] Fix error handling when converting strings to numbers

- Use boost-backed StringUtil's stringTo template function to
reliably convert strings to number types. Error handling follows
the same approach of Qt's QString::toInt(), using an optional
argument to hold the result status for the conversion operation.

[SPEC-1911] Qt-less WAM: replace qt basic data structures

Signed-off-by: Nick Diego Yamane <nickdiego@igalia.com>

nickdiego committed:
[agl][qtless] Use unordered_map instead of std::map

Since at this point no QObject is used as std::map key,
it's safe and easier fully migrate to more efficient
std::unordered_{map,multimap}.

Signed-off-by: Nick Diego Yamane <nickdiego@igalia.com>

jdapena commented on common.pri:
```diff
@@ -32,12 +30,11 @@ isEmpty(PREFIX) {
 # Decide MACHINE_NAME and PLATFORM
 
-MACHINE_NAME = $$(MACHINE)
+MACHINE_NAME = "minnowboard"
```
Oops, this needs to be fixed. If AGL is now using cmake, I guess we can avoid the AGL specific changes in qmake?

antiapuentes replied:
Sure. We can forget about this patch.

jdapena commented on files/launch/WebAppMgr.env:
```diff
@@ -0,0 +1,198 @@
+#fi
+
+#exec $WAM_EXE_PATH $WAM_SWITCHES $WAM_EXTRA_SKIA_CACHE_SWITCHES $WAM_EXTRA_GPU_TUNING_SWITCHES $WAM_WATCHDOG_RENDER_TIMEOUT_SWITCHES $WAM_EXTRA_FLAGS $WAM_V8_CODE_CACHE_SWITCHES --js-flags="$WAM_JS_FLAGS"
+
```
We need some cleanup here: remove all the comments, then move the launch files for AGL to an independent folder.

jdapena commented on files/launch/WebAppMgr.service:
```diff
@@ -0,0 +1,27 @@
+RestartSec=50
+
+[Install]
+WantedBy=default.target
```
Yes, as said, it'd be better to move this to an independent folder. I.e. files/launch/agl/...

jaragunde committed:
[agl] Explicitly link WAM binary with ilmCommon library

Fixes error:
libWebAppMgr.so: undefined reference to `ilm_init'

This seems to be required in master/halibut.

AGL-bug: SPEC-2532

Signed-off-by: Jacobo Aragunde Pérez <jaragunde@igalia.com>

jaragunde committed:
[agl] Explictly link WAM binary with WAMCore lib

Fixes error:
ld: warning: libWebAppMgrCore.so.1.0, needed by src/libWebAppMgr.so.1.0.0, not found (try using -rpath or -rpath-link)
ld: src/libWebAppMgr.so.1.0.0: undefined reference to `WebPageBase::handleLoadStarted()'

AGL-bug: SPEC-2532

Signed-off-by: Jacobo Aragunde Pérez <jaragunde@igalia.com>

antiapuentes committed:
[agl] Remote debugging should work only on debug mode

AGL-bug: SPEC-2327

Signed-off-by: Antia Puentes <apuentes@igalia.com>

antiapuentes committed:
[agl] Update the path for the Remote Web Inspector

AGL-bug: SPEC-2328

jaragunde committed:
Create socket and lock files in XDG_RUNTIME_DIR.

This is required by the new AGL configuration, which runs wam on a
per-user basis.

Bug-AGL: SPEC-2684

Signed-off-by: Jacobo Aragunde Pérez <jaragunde@igalia.com>

antiapuentes committed:
[agl] Suspend/resume webapps depending on visibility

WAM is not receiving the Active/Inactive events from the Window
Manager when the webapp becomes visible/invisible so, react to the
Visible/Invisible events to resume/suspend the webapp instead.

AGL-bug: SPEC-1948

Signed-off-by: Antia Puentes <apuentes@igalia.com>

jaragunde committed:
[agl] Implement an envvar for launcher to wait for host process

When the envvar WAIT_FOR_HOST_SERVICE is set to "1", the launcher
process will wait for the shared WAM service to be enabled, instead of
attempting to run in single browser process mode.

This will be useful when webapps are auto-started on init, which could
mean that the launcher starts before the shared WAM service is ready.

For this to work, we will have to add the envvar to the corresponding
.service files for webapps, which are generated by the app framework
when a .wgt is installed.

AGL-bug: SPEC-2793

Signed-off-by: Jacobo Aragunde Pérez <jaragunde@igalia.com>
